astropy/tests/helper.py

```diff
@@ -222,9 +222,11 @@ def treat_deprecations_as_exceptions():
     for module in list(sys.modules.values()):
         # We don't want to deal with six.MovedModules, only "real"
         # modules. FIXME: we no more use six, this should be useless ?
-        if (isinstance(module, types.ModuleType) and
-                hasattr(module, '__warningregistry__')):
-            del module.__warningregistry__
+        if isinstance(module, types.ModuleType):
```

Comment:
Given the comment, is it perhaps the case that now we are guaranteed that everything in `sys.modules` is of `types.ModuleType`? I.e., could one remove this `if` statement? (seems pretty safe even if it isn't given the new try/except!)

Reply:
I agree. Do you, @pllim?

Reply:
I will sleep on this and reply tomorrow. I cannot brain no more.

Reply:
Yes, it is now removed as per discussions at #12040